util/movement-algs/src/grouping_heuristic/skip.rs

```diff
@@ -0,0 +1,104 @@
+use crate::grouping_heuristic::{GroupingHeuristic, GroupingOutcome};
+
+pub struct Skip<T>(pub Box<dyn GroupingHeuristic<T>>);
+
+impl<T> Skip<T> {
+	pub fn new(heuristic: Box<dyn GroupingHeuristic<T>>) -> Self {
+		Skip(heuristic)
+	}
+
+	pub fn boxed(heuristic: Box<dyn GroupingHeuristic<T>>) -> Box<Self> {
+		Box::new(Skip(heuristic))
+	}
+
+	pub fn skip(
+		&self,
+		distribution: Vec<GroupingOutcome<T>>,
+	) -> Result<Vec<GroupingOutcome<T>>, anyhow::Error> {
+		Ok(distribution)
+	}
+
+	pub fn apply(
+		&mut self,
+		distribution: Vec<GroupingOutcome<T>>,
+	) -> Result<Vec<GroupingOutcome<T>>, anyhow::Error> {
+		self.0.distribute(distribution)
+	}
+}
+
+impl<T> GroupingHeuristic<T> for Skip<T> {
```

Comment:
I'd tried this approach before, but struggled a bit with the unconstrained type `T`. Let me wrangle it.

Reply:
Yeah, I don't think there's an easy and good way to jiggle these types. The stack setup to me either dictates a dynamic dispatch or an enum, because the members are supposed to be different structs.